packages/app/src/pages/transaction/index.tsx

```diff
@@ -96,57 +132,55 @@ export const Transaction: React.FC = () => {
     }
   };
 
-  const setupWithState = async (reqState: ContractCallPayload) => {
-    const source = await client.fetchContractSource({
-      contractName: reqState.contractName,
-      contractAddress: reqState.contractAddress,
-    });
-    setContractSrc(source);
-    setRequestState(reqState);
+  const setupWithState = async (tx: ContractCallPayload | ContractDeployPayload) => {
+    if (tx.txType === 'contract-call') {
+      const contractSource = await client.fetchContractSource({
+        contractName: tx.contractName,
+        contractAddress: tx.contractAddress,
+      });
+      setContractSrc(contractSource);
+      setPendingTransaction(tx);
+    }
+    if (tx.txType === 'contract-deploy') {
```

Comment:
In every other instance I've seen, tx types are denoted as [`contract_call`](https://github.com/blockstack/stacks-blockchain-sidecar/blob/master/docs/entities/transactions/transaction-2-contract-call.schema.json#L16). Where does this value come from that is hyphenated?

Reply:
A string literal type, `TransactionType` is available in the types package, v0.0.13: https://github.com/blockstack/stacks-blockchain-sidecar/pull/73

Reply:
perfect, thank you. I noticed it has been named differently in a few places. Very helpful!